pkg/repository/chartmuseum.go

```diff
@@ -0,0 +1,375 @@
+/*
+Copyright 2023 The Kubebb Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package repository
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ghodss/yaml"
+	"github.com/go-logr/logr"
+	hrepo "helm.sh/helm/v3/pkg/repo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kubebb/core/api/v1alpha1"
+)
+
+const minIntervalSeconds = 120
+
+var _ IWatcher = (*chartmuseum)(nil)
+
+func NewChartmuseum(
+	ctx context.Context,
+	logger logr.Logger,
+	c client.Client,
+	instance *v1alpha1.Repository,
+	statusLen int,
+	cancel context.CancelFunc) IWatcher {
+
+	duration := time.Duration(minIntervalSeconds) * time.Second
+	if instance.Spec.PullStategy != nil && instance.Spec.PullStategy.IntervalSeconds > minIntervalSeconds {
+		duration = time.Duration(instance.Spec.PullStategy.IntervalSeconds) * time.Second
+	}
+
+	if r := time.Duration(time.Second * minIntervalSeconds); r.Milliseconds() > duration.Milliseconds() {
+		logger.Info("the minimum cycle period is 120 seconds, but it is actually less, so the default 300 seconds is used as the period.")
+		duration = r
+	}
+	return &chartmuseum{
+		instance:  instance,
+		c:         c,
+		ctx:       ctx,
+		logger:    logger,
+		duration:  duration,
+		cancel:    cancel,
+		statusLen: statusLen,
+	}
+
+}
+
+type chartmuseum struct {
+	ctx       context.Context
+	cancel    context.CancelFunc
+	instance  *v1alpha1.Repository
+	duration  time.Duration
+	statusLen int
+
+	c      client.Client
+	logger logr.Logger
+}
+
+func (c *chartmuseum) Start() {
+	c.logger.Info("Start to fetch")
```

Comment:
print repository name as well

Reply:
```shell
1.6853264681814563e+09  INFO    Start to fetch  {"controller": "repository", "controllerGroup": "core.kubebb.k8s.com.cn", "controllerKind": "Repository", "repository": {"name":"def","namespace":"kube-system"}, "namespace": "kube-system", "name": "def", "reconcileID": "227ca5df-08ed-4221-8b7b-7b0166ed327e", "Request.Name": "def"}
```
I think there is enough information to describe the object, so it can be done without printing the name.